weasyl/middleware.py

```diff
@@ -19,6 +23,12 @@
 from weasyl.error import WeasylError
 
 
+def _get_headers(env):
+    return {
+        key[5:].replace('_', '-').title(): value
+        for key, value in env.iteritems() if key.startswith('HTTP_')}
```

Comment:
God it makes me sad that `web.py` doesn't have this. I was _pretty_ sure it existed somewhere else in the project already, because `oauthlib` needed the same thing.